packages/build/src/core/report_metrics.ts

```diff
@@ -0,0 +1,31 @@
+import { closeClient, formatTags, InputStatsDOptions, startClient, validateStatsDOptions } from '../report/statsd.js'
+
+export type Metric = {
+  type: 'increment'
+  name: string
+  value: number
+  tags: Record<string, string | string[]>
+}
+
+export type Metrics = Array<Metric>
```

Comment:
nit: I think using `Metrics[]` when you need an array of metrics is totally fine, making this type a bit redundant.

Reply:
Yes that might feel a bit cleaner!